packages/dd-trace/src/log.js

```diff
@@ -40,99 +16,57 @@ const memoize = func => {
   return memoized
 }
 
-function processMsg (msg) {
-  return typeof msg === 'function' ? msg() : msg
-}
-
-function withNoop (fn) {
-  const store = storage.getStore()
-
-  storage.enterWith({ noop: true })
-  fn()
-  storage.enterWith(store)
-}
-
 const log = {
-  _isLogLevelEnabled (level) {
-    return _logLevels[level] >= this._logLevel
+  _isLogLevelEnabled (logLevel) {
+    return logLevel >= this._logLevel
   },
 
-  use (logger) {
-    if (logger && logger.debug instanceof Function && logger.error instanceof Function) {
-      this._logger = logger
+  _publish (logChannel, message) {
+    if (this._isLogLevelEnabled(logChannel.logLevel)) {
+      logChannel.publish(message)
```

Comment:
Seems to me like it would make more sense to _always_ publish and use log levels on the other end to turn subscribers on/off.

Reply:
But then `log.js` should publish log level changes in order to subscribers to update their configuration

Reply:
Why would it need to publish log levels? The decision for log levels should be on the consuming end, shouldn't it? The consumer would just map some log level configuration to meaning "I need to subscribe to X set of channels" and manage that subscription state on their own. 🤔

Reply:
Ok, now log `writer` subscribes to log channels according to its configured log level and `log` always publishes the messages through the channels regardless of the log level

Reply:
With the last commit, `log` only publishes messages if channel has subscribers.